pycbc/workflow/datafind.py

```diff
@@ -496,23 +496,61 @@ def setup_datafind_runtime_cache_single_call_perifo(cp, scienceSegs, outputDir,
     logging.info("Querying datafind server for all science segments.")
     for ifo, scienceSegsIfo in scienceSegs.items():
         observatory = ifo[0].upper()
-        frameType = cp.get_opt_tags("workflow-datafind",
-                                "datafind-%s-frame-type" % (ifo.lower()), tags)
-        # This REQUIRES a coalesced segment list to work
-        startTime = int(scienceSegsIfo[0][0])
-        endTime = int(scienceSegsIfo[-1][1])
-        try:
-            cache, cache_file = run_datafind_instance(cp, outputDir, connection,
-                                       observatory, frameType, startTime,
-                                       endTime, ifo, tags=tags)
-        except:
-            connection = setup_datafind_server_connection(cp, tags=tags)
-            cache, cache_file = run_datafind_instance(cp, outputDir, connection,
-                                       observatory, frameType, startTime,
-                                       endTime, ifo, tags=tags)
-
-        datafindouts.append(cache_file)
-        datafindcaches.append(cache)
+        checked_times = segments.segmentlist([])
+        frame_types = cp.get_opt_tags(
+            "workflow-datafind",
+            "datafind-%s-frame-type" % (ifo.lower()), tags
+        )
+        # Check if this is one type, or time varying
+        frame_types = frame_types.replace(' ', '').strip().split(',')
+        for ftype in frame_types:
+            # Check the times, default to full time initially
+            # This REQUIRES a coalesced segment list to work
+            start = int(scienceSegsIfo[0][0])
+            end = int(scienceSegsIfo[-1][1])
+            # Then check for limits
+            if '[' in ftype:
+                bopt = ftype.split('[')[1].split(']')[0]
+                newstart, newend = bopt.split(':')
+                start = max(int(newstart), start)
+                end = min(int(newend), end)
+                if end <= start:
+                    continue
+                ftype = ftype.replace('[' + bopt +']', '')
```

Comment:
`ftype.split('[')[0]` may be easier to read?

Reply:
Makes sense (I was using the DQ code as a model, but there the syntax is a bit more involved). Changed.